sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePlugin.scala

```diff
@@ -14,8 +14,13 @@ object ScalaNativePlugin extends AutoPlugin {
 
     val nativeVersion = nir.Versions.current
 
+    val nativeBuildConfig =
+      taskKey[build.Config]("Full build configuration, Config")
+
```

Comment:
`nativeBuildConfig` should probably be hidden from the users not to allow for its modification. MainClass and classPath shall always be up to date with the values coming from `selectMainClass` or other classes. Similarly to classpath which is managed by the sbt.
Workdir contains mostly internal data - *.hnir, *.ll, and *.o files. These should be kept away from the users and shall be managed only the build tool.
The only benefit of changing the workdir is to change the output directory for the result program file. I think that we can add a dedicated setting in the `NativeConfig` for the output directory where the final executable file would be moved upon successful linkage.

To make this task hidden we should move it to the `ScalaNativePluginInternal`

Reply:
Sure, I thought it was nice to be able to view it, but it does open it up for users so we probably don't want that. I think a way to copy the executable somewhere else might be a good feature if we get demand and some good use cases.

Reply:
Shouldn't the `tasks` in the plugin theoretically be `settings`.

Reply:
I think I have addressed the review comments now.